server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -143,6 +142,10 @@ private static KeywordFieldMapper toType(FieldMapper in) {
         return (KeywordFieldMapper) in;
     }
 
+    public static Builder buildForTest(final String name, boolean hasDocValues) {
```

Comment:
I don't think this is needed and is unrelated to adding the new index setting? We can keep using the constructor  with two parameters?

Reply:
I removed this after changing the available constructors.